pkg/system/phase4_configuring.go

```diff
@@ -510,6 +510,17 @@ func (r *Reconciler) generateBackingStoreTargetName() string {
 	if len(name) > MaxNameLength {
 		name = name[:MaxNameLength]
```

Comment:
I think this is not what we want. Instead of truncating the end of the route name, we should prefer to truncate its beginning and keep the end because in domains this is usually the more identifying part.
So I would change this flow to something like this:
```golang
	name := "nb." + tsMilli
	if r.RouteMgmt.Spec.Host != "" {
		suffix := ""
		hostItems := strings.Split(r.RouteMgmt.Spec.Host, ".")
		for i := len(hostItems) - 1; i >= 0; i-- {
			hostItem := strings.Trim(hostItems[i], "-.")
			if len(name) + 1 + len(hostItem) + 1 + len(suffix) > 63 {
				break
			}
			suffix = hostItem + "." + suffix
		}
		name += "." + suffix
	}
```

Reply:
you're right. I'll change